From 8af86e58356af8b9e7c8d3a996cf62169f96f58f Mon Sep 17 00:00:00 2001
From: Nikolay Aleksandrov <nikolay@redhat.com>
Date: Mon, 1 Sep 2014 13:33:09 -0400
Subject: [crypto] qat: remove an unneeded cast

Message-id: <1409578401-26943-18-git-send-email-nikolay@redhat.com>
Patchwork-id: 90093
O-Subject: [RHEL7.1 PATCH BZ 1091130 17/29] crypto: qat - remove an unneeded cast
Bugzilla: 1091130
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Herbert Xu <herbert.xu@redhat.com>

From: Dan Carpenter <dan.carpenter@oracle.com>

The cast to (unsigned int *) doesn't hurt anything but it is pointless.

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
(cherry picked from commit ac1a2b49ea3dff611d820043a0c597db3867bee0)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/crypto/qat/qat_common/qat_hal.c b/drivers/crypto/qat/qat_common/qat_hal.c
index 903ca88..28da876 100644
--- a/drivers/crypto/qat/qat_common/qat_hal.c
+++ b/drivers/crypto/qat/qat_common/qat_hal.c
@@ -850,7 +850,7 @@ static int qat_hal_exec_micro_inst(struct icp_qat_fw_loader_handle *handle,
 
  if ((inst_num > handle->hal_handle->max_ustore) || !micro_inst) {
   pr_err("QAT: invalid instructs inst_num=%d, micro_inst=0x%p\n ",
-         inst_num, (unsigned int *)micro_inst);
+         inst_num, micro_inst);
   return -EINVAL;
  }
  /* save current context */
-- 
1.7.1